### DISCUSSION TITLE
Pyinstaller problems

### DISCUSSION THREAD

**Discussion by frankenstein91** [zauberzeug/nicegui#1141] [Q&A] [answered]
### Question

First, I share the "minimal" source code to recreate the problem.

``` Python3
import argparse
import multiprocessing
from nicegui import ui

def main():
    parser = argparse.ArgumentParser(description="This is only a demo for the problem")
    parser.add_argument("-p", "--port", type=int, help="set server port", default=7780)
    args = parser.parse_args()
    ui.run(port=args.port, title="Problem Demo")

if __name__ in {"__main__", "__mp_main__"}:
    multiprocessing.freeze_support()
    main()
```
I have tried to convert my script using `pyinstaller -F --clean -y --collect-all nicegui --hiddenimport nicegui test1.py` into a single file, but somehow get stuck at this point.

```
./dist/test1 
ERROR:    Traceback (most recent call last):
  File "starlette/routing.py", line 677, in lifespan
  File "starlette/routing.py", line 566, in __aenter__
  File "starlette/routing.py", line 656, in startup
  File "nicegui/nicegui.py", line 62, in handle_startup
    raise RuntimeError('\n\n'
RuntimeError: 

You must call ui.run() to start the server.
If ui.run() is behind a main guard
   if __name__ == "__main__":
remove the guard or replace it with
   if __name__ in {"__main__", "__mp_main__"}:
to allow for multiprocessing.

ERROR:    Application startup failed. Exiting.
```

I have a Venv with this installed
aiofiles==23.1.0
altgraph==0.17.3
anyio==3.7.0
bidict==0.22.1
black==23.3.0
bottle==0.12.25
certifi==2023.5.7
charset-normalizer==3.1.0
click==8.1.3
contourpy==1.0.7
cycler==0.11.0
docopt==0.6.2
fastapi==0.96.0
fastapi-socketio==0.0.10
fonttools==4.39.4
h11==0.14.0
httptools==0.5.0
idna==3.4
importlib-metadata==6.6.0
isort==5.12.0
itsdangerous==2.1.2
Jinja2==3.1.2
kiwisolver==1.4.4
markdown2==2.4.8
MarkupSafe==2.1.3
matplotlib==3.7.1
mypy-extensions==1.0.0
names==0.3.0
nicegui==1.2.17
numpy==1.24.3
orjson==3.9.0
packaging==23.1
pathspec==0.11.1
Pillow==9.5.0
pipreqs==0.4.13
platformdirs==3.5.1
plotly==5.14.1
proxy-tools==0.1.0
pscript==0.7.7
pydantic==1.10.9
Pygments==2.15.1
pyinstaller==5.12.0
pyinstaller-hooks-contrib==2023.3
pyparsing==3.0.9
python-dateutil==2.8.2
python-dotenv==1.0.0
python-engineio==4.4.1
python-multipart==0.0.6
python-socketio==5.8.0
pytz==2023.3
pywebview==4.1
PyYAML==6.0
requests==2.31.0
six==1.16.0
sniffio==1.3.0
starlette==0.27.0
tenacity==8.2.2
typing_extensions==4.6.3
urllib3==2.0.3
uvicorn==0.20.0
uvloop==0.17.0
vbuild==0.8.1
watchfiles==0.18.1
websockets==11.0.3
yamlmaker==0.0.8
yarg==0.1.9
zipp==3.15.0

**Comment by falkoschindler** [comment, marked as answer]
Hi @frankenstein91!
Please have a look at the documentation about packaging a NiceGUI app: https://nicegui.io/documentation#package_for_installation
It looks like you didn't set `reload=False`, which might cause the runtime error.

**Reply by frankenstein91** [reply]
one problem ramains on exit...
```
^CERROR:    Traceback (most recent call last):
  File "starlette/routing.py", line 686, in lifespan
  File "uvicorn/lifespan/on.py", line 137, in receive
  File "asyncio/queues.py", line 158, in get
asyncio.exceptions.CancelledError
```

**Reply by falkoschindler** [reply]
Ok, let's continue discussing the `CancelledError` over at https://github.com/zauberzeug/nicegui/discussions/1153.
